projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -55,6 +76,7 @@ protected function __construct() {
 			add_filter( 'admin_menu', array( $this, 'override_svg_icons' ), 99999 );
 			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ), 11 );
 			add_action( 'admin_head', array( $this, 'set_site_icon_inline_styles' ) );
+			add_action( 'admin_menu', array( $this, 'handle_preferred_view' ), 99997 );
```

Comment:
Would moving this out of the `! $this->is_api_request`, help us set the `preferred-view` when being redirected from WP Admin to Calypso with `?preferred-view=default&origin={slug}`? In that case Calypso would send the `admin-menu` request along with `preferred-view=default` & `origin={slug}`?

At any case, we can handle this in a follow up.

Reply:
Yup, it'd depend on how we'll handle the WP Admin > Calypso redirect. I was thinking on a different approach rather than using the API:
- Users clicks on the WP Admin > Calypso switcher.
- We make an AJAX call (e.g. `admin-ajax.php?screen=index.php&preferred-view=default`) to store the new preference.
- Once the AJAX call finishes, we redirect the user to Calypso.

Reply:
Cool, that's certainly a better way!